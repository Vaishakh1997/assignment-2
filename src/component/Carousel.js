import React from 'react'
import '../App.css'
import store from '../store'
import { keyframes, css } from 'styled-components'

function Carousel({images}){
    const totalImages = 100/images.length

    let totalTansition = images.map((image, index) => {
        return `${totalImages * index}%  { transform: translateX(-${100*index}%)}` 
    })

    let slideShow = keyframes`
    ${totalTansition} 100s ease-out 1s infinite
    `

    // const animation = css`animation: ${slideShow} 100s ease-out 1s infinite`

        let imageSlider = images.map((image, index) => {
            return (
                <div key={index} className="slide" style={{ animation: {slideShow}}} key={index}>
                    <img style={{width:'100%',height:'100%'}} src={image} alt="1"></img>
                </div>
            )
        })
        console.log(slideShow)


        return (
            <div className="slider">{imageSlider}</div>
        )
    }
export default Carousel;