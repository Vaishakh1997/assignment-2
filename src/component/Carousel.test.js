import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Carousel from './Carousel';

const mockStore = configureMockStore();
const props= { images: [1,2] }
const store = mockStore({});

let wrapped = shallow(
    <Provider store={store}>
        <Carousel {...props}/>
    </Provider>
)


describe("Albums Component", () => {

    it("Albums should render without throwing an error", () => {
        expect(wrapped.debug()).toMatchSnapshot();
    });
    
    it('Expected props loading as false', () => {
        expect(wrapped.dive().props().images).toEqual(props.images)
    });
    
});