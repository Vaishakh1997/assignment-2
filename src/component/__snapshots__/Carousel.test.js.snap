// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Carousel should render without throwing an error 1`] = `ShallowWrapper {}`;
